react-components/toolbar-filter/step2.md

```diff
@@ -1,61 +1,141 @@
+
+1) **Locate the following constructor in `ToolbarDemoApp**
+
+<pre class="file">
+constructor(props) {
+  super(props);
+}
+</pre>
+
+2) **Initialize the state in the constructor located in step 1**
+
+Add a `filters` object to the state. `filters` should have three properties (name, risk, and status) each with an empty array as its value. This filters object will manage the three filters in the `DataToolbar`.
+
+Once step 2 is completed, the constructor should look like the snippet below.
+
+<pre class="file">
+constructor(props) {
+  super(props);
+  this.state = {
+    filters: {
+      name: [],
+      risk: [],
+      status: []
+    }
+  };
+}
+</pre>
+
+3) **Add two event handlers to the `ToolbarDemoApp` class**
+
+Both of the event handlers can be inserted outside and below the constructor modified in the previous steps.
+
+The first will be an `onSearch` event handler which will fire when the user interacts with the name filter.
+
+<pre class="file" data-target="clipboard">
+onSearch = (value) =&gt; {
+  this.setState(prevState =&gt; {
+    return {
+      filters: {
+        ...prevState.filters,
+        name: [...prevState.filters.name, value]
+      }
+    };
+  });
+}
+</pre>
+
+The second event handler will be an `onSelect` event handler which will fire when the user interacts with the risk and status filters.
+
+<pre class="file" data-target="clipboard">
+onSelect = (type, event, selection) =&gt; {
+  const checked = event.target.checked;
+  this.setState(prevState =&gt; {
+    const prevSelections = prevState.filters[type.toLowerCase()];
+    return {
+      filters: {
+        ...prevState.filters,
+        [type.toLowerCase()]: checked
+          ? [...prevSelections, selection]
+          : prevSelections.filter(value => value !== selection)
+      }
+    };
+  });
 };
+</pre>
+
+4) **Add the following constants inside the top of the render function above the return statement**
+
+In this case, unwrap the state managed `filters` to be used by the `DataToolbar`.
+
+Also define the status and risk options that will appear in the two dropdown filters in the `DataToolbar`.
 
-export default ToolbarDemoApp;
+<pre class="file" data-target="clipboard">
+const { filters } = this.state;
+const statusOptions = [&quot;New&quot;, &quot;Pending&quot;, &quot;Running&quot;, &quot;Cancelled&quot;];
+const riskOptions = [&quot;Low&quot;, &quot;Medium&quot;, &quot;High&quot;];
 </pre>
 
-3) Each toolbar must contain one or more content rows. Wrap the "Toolbar Demo" text in a `DataToolbarContent` component and place that `DataToolbarContent` into a `DataToolbar`.
+
+5) **Locate the following `DataToolbarContent` component:**
+
+<pre class="file">
+&lt;DataToolbar id=&quot;tutorial-toolbar&quot;&gt;
+  &lt;DataToolbarContent&gt;
+    Toolbar Demo
+  &lt;/DataToolbarContent&gt;
+&lt;/DataToolbar&gt;
+</pre>
+
+6) **Replace the `Toolbar Demo` text in the `DataToolbarContent` with the following six items:**
```

Comment:
I think you should add a sentence here that explains briefly what six items are being added